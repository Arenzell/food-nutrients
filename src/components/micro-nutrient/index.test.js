import React from 'react'
import MicroNutrient from './index'
import renderer from 'react-test-renderer'

it('renders correctly with data', () => {
  const microNutrient = {
    name: 'Vitamin A',
    wiki: 'https://en.wikipedia.org/wiki/Vitamin_A',
  }
  const microNutrientData = {
    raw: 50,
    amount: 5,
    percentage: 30,
    unit: 'mg',
  }
  const tree = renderer
    .create(<MicroNutrient microNutrient={microNutrient} microNutrientData={microNutrientData} />)
    .toJSON()
  expect(tree).toMatchSnapshot()
})

it('renders correctly with null data', () => {
  const microNutrient = {
    name: 'Vitamin A',
    wiki: 'https://en.wikipedia.org/wiki/Vitamin_A',
  }
  const microNutrientData = null
  const tree = renderer
    .create(<MicroNutrient microNutrient={microNutrient} microNutrientData={microNutrientData} />)
    .toJSON()
  expect(tree).toMatchSnapshot()
})
