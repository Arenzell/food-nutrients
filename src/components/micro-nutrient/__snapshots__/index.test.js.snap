// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly with data 1`] = `
<div
  className="microNutrient"
>
  <div
    className="microNutrientName"
  >
    <a
      href="https://en.wikipedia.org/wiki/Vitamin_A"
      rel="noopener noreferrer"
      target="_blank"
    >
      Vitamin A
    </a>
  </div>
  <div
    className="microNutrientProgressBar"
  >
    <div
      className="ant-progress ant-progress-line ant-progress-status-normal ant-progress-small"
    >
      <div>
        <div
          className="ant-progress-outer"
        >
          <div
            className="ant-progress-inner"
          >
            <div
              className="ant-progress-bg"
              style={
                Object {
                  "background": undefined,
                  "borderRadius": "100px",
                  "height": 6,
                  "width": "30%",
                }
              }
            />
          </div>
        </div>
      </div>
    </div>
    <span>
      30
      % 
    </span>
  </div>
  <div
    className="microNutrientAmount"
  >
    5
     
    mg
  </div>
</div>
`;

exports[`renders correctly with null data 1`] = `
<div
  className="microNutrient"
>
  <div
    className="microNutrientName"
  >
    <a
      href="https://en.wikipedia.org/wiki/Vitamin_A"
      rel="noopener noreferrer"
      target="_blank"
    >
      Vitamin A
    </a>
  </div>
  <div
    className="microNutrientProgressBar"
  >
    <div
      className="ant-progress ant-progress-line ant-progress-status-normal ant-progress-small"
    >
      <div>
        <div
          className="ant-progress-outer"
        >
          <div
            className="ant-progress-inner"
          >
            <div
              className="ant-progress-bg"
              style={
                Object {
                  "background": undefined,
                  "borderRadius": "100px",
                  "height": 6,
                  "width": "0%",
                }
              }
            />
          </div>
        </div>
      </div>
    </div>
    <span>
      0
      % 
    </span>
  </div>
  <div
    className="microNutrientAmount"
  >
    0
     
    Î¼g
  </div>
</div>
`;
